lib/internal/modules/typescript.js

```diff
@@ -87,11 +92,18 @@ function stripTypeScriptTypes(code, options = kEmptyObject) {
   });
 }
 
+/**
+ * @typedef {object} TypeScriptOptions
+ * @property {'transform'|'strip-only'} mode Mode.
```

Comment:
Nit, so you can avoid repeating `'strip-only'|'transform'` later

```suggestion
 * @typedef {'strip-only' | 'transform'} TypeScriptMode
 *
 * @typedef {object} TypeScriptOptions
 * @property {TypeScriptMode} mode Mode.
```